game-app/game-core/src/main/java/games/strategy/triplea/ui/EditPanel.java

```diff
@@ -480,68 +483,95 @@ public void actionPerformed(final ActionEvent event) {
             // continued in territorySelected() handler below
           }
         };
-    changePUsAction =
-        new AbstractAction("Change PUs") {
+    changeResourcesAction =
+        new AbstractAction("Change Resources") {
           private static final long serialVersionUID = -2751668909341983795L;
 
           @Override
           public void actionPerformed(final ActionEvent event) {
             currentAction = this;
             setWidgetActivation();
-            final PlayerChooser playerChooser =
-                new PlayerChooser(getData().getPlayerList(), getMap().getUiContext(), false);
-            final JDialog dialog =
-                playerChooser.createDialog(getTopLevelAncestor(), "Select owner PUs to change");
-            dialog.setVisible(true);
-            final GamePlayer player = playerChooser.getSelected();
-            if (player == null) {
+
+            final Optional<GamePlayer> player = choosePlayer();
+            if (player.isEmpty()) {
               cancelEditAction.actionPerformed(null);
               return;
             }
-            getData().acquireReadLock();
-            final Resource pus;
-            try {
-              pus = getData().getResourceList().getResource(Constants.PUS);
-            } finally {
-              getData().releaseReadLock();
+
+            final Optional<Resource> resource = chooseResource();
+            if (resource.isEmpty()) {
```

Comment:
nit, rather than doing:
```
 final Optional<Resource> resource = chooseResource();
   if (resource.isEmpty()) {
         return;
    }
    resource.get()
```

I found it's easier to do this:
```
 final Resource resource = chooseResource().orElse(null);
   if (resource == null) {
         return;
    }
    resource
```

That avoids the need to do a 'resource.get' later. My rule of thumb for optional is one should always avoid passing it as a parameter and secondarily avoid assignment to an option (but it's okay to return an optional, that also signals that all error handling & notifications have been performed).

Reply:
I think you might be able to use the above pattern to avoid quite a number of `.get()` method calls later, WDYT? Worth updating?

Reply:
Thanks, made the change. Agree it's nicer.